presto-main/src/main/java/io/prestosql/operator/scalar/DateTimeFunctions.java

```diff
@@ -145,6 +153,50 @@ public static long fromUnixTime(@SqlType(StandardTypes.DOUBLE) double unixTime,
         return packDateTimeWithZone(Math.round(unixTime * 1000), zoneId.toStringUtf8());
     }
 
+    @ScalarFunction("from_unixtime_nanos")
+    public static final class FromUnixtimeNanosDecimal
+    {
+        private FromUnixtimeNanosDecimal() {}
+
+        @LiteralParameters({"p", "s"})
+        @SqlType("timestamp(9)")
+        public static LongTimestamp fromLong(@LiteralParameter("s") long scale, @SqlType("decimal(p, s)") Slice unixTimeNanos)
+        {
+            BigInteger unixTimeNanosInt = unscaledDecimalToBigInteger(rescale(unixTimeNanos, -(int) scale));
+            long epochSeconds = unixTimeNanosInt.divide(BigInteger.valueOf(NANOSECONDS_PER_SECOND)).longValue();
+            long nanosOfSecond = unixTimeNanosInt.remainder(BigInteger.valueOf(NANOSECONDS_PER_SECOND)).longValue();
+            long picosOfSecond = nanosOfSecond * PICOSECONDS_PER_NANOSECOND;
+            if (picosOfSecond < 0) {
```

Comment:
Add a comment to indicate that this is simulating floorMod/floorDiv due to lack of direct support in BigInteger